app/src/main/java/com/sdp/movemeet/backend/serialization/ActivitySerializer.java

```diff
@@ -70,6 +77,11 @@ public Activity deserialize(Map<String, Object> data) {
                 ((Timestamp) data.get(CREATION_KEY)).toDate()
         );
 
+        Map<String, GPSPath> recordings = (Map<String, GPSPath>) data.getOrDefault(GPS_RECORDINGS_KEY, null);
```

Comment:
No, it's recorded as a GPSPath. The String is the uid of the user to whom the recording belongs (since many users can record paths for the same activity)

Reply:
Oh I see, all clear I understand the working of your `Map` now.